### PR TITLE
fixed the bug `~ | path type` return empty string

### LINKED ISSUE #9849: `'~' | path type` should return 'dir'
### Describe the bug

`'~' | path type` or `mkdir ~/test; '~/test' | path type` should return 'dir', but now it's return empty string.

### How to reproduce

1. 
```nu
use std;
std assert (('~' | path type) == dir) $"'~' is the diretory: ('~' | path expand)"
```

2. 
```nu
use std;
mkdir ~/test; std assert (('~/test' | path type) == dir) $"'~' is the diretory: ('~/test' | path expand)"
```

3.
```nu
use std;
std assert (('~' | path type) == ('~' | path expand | path type) ) 
```

4.
```nu
use std;
 mkdir ~/test; std assert (('~/test' | path type) == ('~/test' | path expand | path type) )
```

### Expected behavior

Test examples 1/2/3/4 should be tested successfully;

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.83.1                                   |
| branch             | main                                     |
| commit_hash        | 6ac3351fd13e317aacebaa8203ed01fc94cc4025 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16)      |
| rust_channel       | 1.69.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12)      |
| build_time         | 2023-07-29 09:40:28 +08:00               |
| build_rust_channel | release                                  |
| allocator          | standard                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mengsuenyan**
- this PR should close #9849 
- fixes #9849

### CODE REVIEW

File: crates/nu-command/tests/commands/path/type_.rs [hunk 1]
```diff
@@ -31,6 +31,18 @@ fn returns_type_of_existing_file() {
...
+        // let actual = nu!(
+        //     cwd: dirs.test(), pipeline(
+        //     r#"
+        //         cd ~; touch spam.txt;
+        //         echo "~/spam.txt"
+        //         | path type
+        //     "#
+        // ));
+
+        // assert_eq!(actual.out, "file");
```
Comment (amtoine) [inline crates/nu-command/tests/commands/path/type_.rs:45]: can we get rid of this commented block of dead code? :yum:

File: crates/nu-command/tests/commands/path/type_.rs [hunk 2]
```diff
@@ -50,5 +62,15 @@ fn returns_type_of_existing_directory() {
...
+
+        let actual = nu!(
+            cwd: dirs.test(), pipeline(
+            r#"
+                echo "~"
+                | path type
+            "#
+        ));
+
+        assert_eq!(actual.out, "dir");
```
Comment (amtoine) [inline crates/nu-command/tests/commands/path/type_.rs:74]: modulo some formatting, you should be able to simplify this test to this :wink: 
```suggestion

        let actual = nu!(r#"
            echo "~" | path type
        "#);

        assert_eq!(actual.out, "dir");
```

**Review by amtoine — DISMISSED** [review]
thanks for tackling this bug :pray:

**Comment by amtoine** [comment]
thanks for the fixes :blush:

**Comment by fdncred** [comment]
Thanks

**Pull request merged**